hpx/parallel/executors/execution_information.hpp

```diff
@@ -270,15 +323,99 @@ namespace hpx { namespace parallel { namespace execution
+                return set_scheduler_mode(std::forward<Executor>(exec), mode);
+            }
+#ifndef HPX_HAVE_CXX11_AUTO
+        };
+#endif
         /// \endcond
     }
+
+    // define customization points
+    namespace
+    {
+        /// Retrieve the number of (kernel-)threads used by the associated
+        /// executor.
+        ///
+        /// \param exec  [in] The executor object to use to extract the
+        ///              requested information for.
+        ///
+        /// \note This calls exec.os_thread_count() if it exists;
+        ///       otherwise it executes hpx::get_os_thread_count().
+        ///
+        constexpr detail::customization_point<
+                detail::processing_units_count_tag
+            > const& processing_units_count = detail::static_const<
+                    detail::customization_point<detail::processing_units_count_tag>
+                >::value;
+
+        /// Retrieve whether this executor has operations pending or not.
+        ///
+        /// \param exec  [in] The executor object to use to extract the
+        ///              requested information for.
+        ///
+        /// \note If the executor does not expose this information, this call
+        ///       will always return \a false
+        ///
+        constexpr detail::customization_point<
+                detail::has_pending_closures_tag
+            > const& has_pending_closures = detail::static_const<
+                    detail::customization_point<detail::has_pending_closures_tag>
+                >::value;
+
+        /// Retrieve the bitmask describing the processing units the given
+        /// thread is allowed to run on
+        ///
+        /// All threads::executors invoke sched.get_pu_mask().
+        ///
+        /// \param exec  [in] The executor object to use for querying the
+        ///              number of pending tasks.
+        /// \param topo  [in] The topology object to use to extract the
+        ///              requested information.
+        /// \param thream_num [in] The sequence number of the thread to
+        ///              retrieve information for.
+        ///
+        /// \note If the executor does not support this operation, this call
+        ///       will always invoke hpx::threads::get_pu_mask()
+        ///
+        constexpr detail::customization_point<detail::get_pu_mask_tag> const&
+            get_pu_mask = detail::static_const<
+                    detail::customization_point<detail::get_pu_mask_tag>
+                >::value;
+
+        /// Set various modes of operation on the scheduler underneath the
+        /// given executor.
+        ///
+        /// \param exec     [in] The executor object to use.
+        /// \param mode     [in] The new mode for the scheduler to pick up
+        ///
+        /// \note This calls exec.set_scheduler_mode(mode) if it exists;
+        ///       otherwise it does nothing.
+        ///
+        constexpr detail::customization_point<
```

Comment:
Yeah, maybe better do that in a separate PR instead of getting sidetracked here :-)